oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/PersistenceTimer.java

```diff
@@ -77,9 +83,16 @@ public void start(ModuleManager moduleManager, CoreModuleConfig moduleConfig) {
             "persistence_timer_bulk_execute_latency", "Latency of the execute stage in persistence timer",
             MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
         );
+        allLatency = metricsCreator.createHistogramMetric(
+            "persistence_timer_bulk_all_latency", "Latency of the all stage in persistence timer",
+            MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
+        );
+
         syncOperationThreadsNum = moduleConfig.getSyncThreads();
         maxSyncoperationNum = moduleConfig.getMaxSyncOperationNum();
+        batchExecutorService = Executors.newSingleThreadExecutor();
```

Comment:
We will use one more thread because of this, besides the new add `moduleConfig#getPrepareThreads()`. Do we have a way to avoid this? Every thread is a very expensive resource in the OAP.

Reply:
Yes. We can avoid this thread by just use multi-consumer. It will bring some competition, but it is ok.
```
            List<Future<?>> batchFutures = new ArrayList<>();
            for (int i = 0; i < syncOperationThreadsNum; i++) {
                Future<?> batchFuture = executorService.submit(() -> {
                    // consume the metrics
                    while (!stop.get()) {
                        List<PrepareRequest> partition = prepareQueue.popMany();
                        if (partition.isEmpty()) {
                            break;
                        }
                        HistogramMetrics.Timer executeLatencyTimer = executeLatency.createTimer();
                        try {
                            if (CollectionUtils.isNotEmpty(partition)) {
                                batchDAO.synchronous(partition);
                            }
                        } catch (Throwable e) {
                            log.error(e.getMessage(), e);
                        } finally {
                            executeLatencyTimer.finish();
                        }
                    }
                    return null;
                });
                batchFutures.add(batchFuture);
            }

```


for (Future<?> result : batchFutures) {
      result.get();
 }